docs/develprocess.md

```diff
@@ -108,6 +108,57 @@ Our policy is to upstream everything if possible. However, there are some except
 ### How to help at development
 It all depends on your skills and areas of interest so it's hard to tell specifically in advance. Having a look at the open github issues and pick one (<https://github.com/xcp-ng/xcp/issues>) could definitely help. Else, maybe there's a specific topic that you would want to help improve. Even if you don't know where to start, just come and talk with us (see "Where discussion happens" above).
 
+## Tags, maintenance branches in our code repositories
+
+Objectives of the branch and tag naming conventions:
+* always know how to name tags and maintenance branches depending on the situation
+* easily identify maintenance branches for a given release of XCP-ng, based on their name
+* know what branch to develop the next version on
+* for our tags and branches that have been developed from upstream branches or tags, document the upstream branch names through our branch and tag naming
+
+First question to ask ourselves: **who is the upstream for the software**?
+
+### 1. We are upstream
+
+Example: `xcp-emu-manager`
+* Tags: `vMAJOR.MINOR.PATCH` (`v1.1.2`, `v1.2.0`...)
+* Maintenance branch if needed: `VERSION-XCPNGVERSION}`.
+  * We don't need to create the maintenance branch in advance. Not all software gets hotfixes.
+  * `VERSION` is the version we branched from:
+    * If possible `MAJOR.MINOR` (`1.1`, `1.2`)
+    * ... Unless we tagged the project in a way that would make this ambiguous. In that case, `MAJOR.MINOR.PATCH` (`1.1.1`, `1.1.2`...).
+  * `XCPNGVERSION` is the two-digit version of XCP-ng: `8.1`, `8.2`...
+  * Examples: `1.1.2-8.0`, `1.2-8.2`... 
+* Next release developed on: `master`
+
+* Tag: `v1.2.0`
+* Maintenance branch: `maint-1.2-8.2`
+
+If for any reason we decide to release a newer version of the software as a maintenance update, then:
+* We stop updating the existing maintenance branch
+* Further hotfixes would come from a new maintenance branch created from the appropriate tag.
+
+Special case: VERSION and XCPNGVERSION are always the same (example: `xcp-ng-release`). Then:
+* Tags: `vXCPNGVERSIONFULL` (`v8.2.0`)
+* Maintenance branch if needed: `XCPNGVERSION` (`8.2`)
+
+### 2. We are downstream
+
+Examples: `host-installer`, `
+* Tags: `vUPSTREAMVERSION-XCPNGVERSIONFULL` (`v1.29.0-8.2.0`)
+* Maintenance branch: `UPSTREAMVERSION-XCPNGVERSION` (`1.29.0-8.2`)
```

Comment:
How d we know our version here?
We're one upstream release `1.29.0` and xcp version is `8.2`, what about we make 2 releases of this packages with bugfixes? The version becomes `1.29.2-8.2`? What if there's an upstream `1.29.2` as well?

Reply:
Good question. We wouldn't modify the version `1.29.0` because versions are defined by upstream. Our version is `1.29.0+patches` actually. If we keep the proposal as is, I suppose any further tag from the maintenance branch would be like `1.29.0-1-8.2` but I don't like it very much. In fact, I don't think we'd tag at all after the initial release. At least that's what I do currently: I tag at the date of the release, but then I only update the maintenance branch without tagging more. Since the branch is mainly for producing patches, not for creating tarballs, I'm not sure tags are strictly required (and sending a build of the RPM to koji is in itself a "form" of tag).

Reply:
Fine for me. :)

Reply:
I've updated the document with information about that.

Reply:
My reading was that maintenance releases contain the bugfixes branched from the previous maintenance release.  So after two bugfixes from `1.29.0-8.2` (say `v1.29.0-8.2.1` and `v1.29.0-8.2.2`) and two new upstream releases, we would have `1.29.2-8.3` with those two bugfixes incorporated.  Is this right?

Reply:
The thing here is that `8.2` is the version of XCP-ng, not of the software itself (`sm`, in our example). So we can't increment it because an update of that package would not mean a new XCP-ng 8.2.1 release. In fact, I could have replaced `8.2` with `xcp-ng` in the branch name, but we'd lose the information about the XCP-ng release this is about.

The situation is easier when we're the upstream and can increment the version of the software.